src/App.jsx

```diff
@@ -1,26 +1,29 @@
-import { CacheProvider } from "@emotion/react";
-import { CssBaseline } from "@mui/material";
-import { ThemeProvider } from "@mui/material/styles";
-import React from "react";
-import { CredentialsProvider } from "./hooks/use-credentials";
-import { AppRouter } from "./routes";
-import { theme } from "./theme";
-import { createEmotionCache } from "./utils/create-emotion-cache";
-import { ConfigProvider } from "@4chain-ag/react-configuration";
+import { CacheProvider } from '@emotion/react';
+import { CssBaseline } from '@mui/material';
+import { ThemeProvider } from '@mui/material/styles';
+import React from 'react';
+import { CredentialsProvider } from './hooks/useCredentials';
+import { UserProvider } from './hooks/useUser';
+import { AppRouter } from './routes';
+import { theme } from './theme';
+import { createEmotionCache } from './utils/create-emotion-cache';
+import { ConfigProvider } from '@4chain-ag/react-configuration';
 
 const clientSideEmotionCache = createEmotionCache();
 
 export const App = () => {
   return (
     <CacheProvider value={clientSideEmotionCache}>
-        <ConfigProvider>
-          <CredentialsProvider>
+      <ConfigProvider>
+        <CredentialsProvider>
+          <UserProvider>
```

Comment:
For now on, the `spvWalletClient` is created in `UserProvider` once and shared via a context. 

We should not create local clients in other places of code (with one exception: `login.js`)